### PR TITLE
Functional test infrastructure

### PR DESCRIPTION
**Pull request by Tratcher**
Here is the infrastructure and a few proof-of-concept functional test.

@prabu, @muratg

### CODE REVIEW

File: NuGet.master.config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
```
Comment (Praburaj) [inline NuGet.master.config]: why do we need these?

File: global.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "projects": ["src"]
```
Comment (Praburaj) [inline global.json]: nit: format this file.

File: test/ServerComparison.TestSites/StartupHelloWorld.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Runtime;
+
+namespace ServerComparison.TestSites
+{
+    /// <summary>
+    /// To make runtime to load an environment based startup class, specify the environment by the following ways:
+    /// 1. Drop a Microsoft.AspNet.Hosting.ini file in the wwwroot folder
+    /// 2. Add a setting in the ini file named 'ASPNET_ENV' with value of the format 'Startup[EnvironmentName]'. For example: To load a Startup class named
+    /// 'StartupHelloWorld' the value of the env should be 'NtlmAuthentication' (eg. ASPNET_ENV=HelloWorld). Runtime adds a 'Startup' prefix to this and loads 'StartupHelloWorld'.
```
Comment (Praburaj) [inline test/ServerComparison.TestSites/StartupHelloWorld.cs]: Fix all comments (`NtlmAuthentication` -> `Helloworld`)or remove this. I think this is repeated in a number of places.

**Comment by Praburaj** [comment]
Looks good to me. Other than those minor comments :shipit:

File: NuGet.master.config (hunk 1 above)
Comment (Tratcher) [inline NuGet.master.config]: Not sure, it's part of the build template.
https://github.com/aspnet/Universe/tree/dev/build-template

**Pull request merged**